README.md

```diff
@@ -1,170 +1,60 @@
+eof
 ```
-
-### Example: using the defaults
-
-To use the defaults, just call `setup()` with an empty `config` parameter.
-For the `gopls` config, that would be:
-
-```lua
-require'lspconfig'.gopls.setup{}
+3. Create a project, this project must contain a file matching the root directory trigger. See [Automatically launching language servers](#Automatically-launching-language-servers) for additional info.
+```bash
+mkdir test_python_project
+cd test_python_project
+git init
+touch main.py
 ```
+4. Launch neovim, the language server will now be attached and providing diagnostics (see `:LspInfo`)
+`nvim main.py`
+5. See [Keybindings and completion](#Keybindings-and-completion) for mapping useful functions and enabling omnifunc completion
 
-### Example: override some defaults
+## Automatically launching language servers
 
-To set some config properties at `setup()`, specify their keys. For example to
-change how the "project root" is found, set the `root_dir` key:
+In order to automatically launch a language server, lspconfig searches up the directory tree from your current buffer to find a file matching the `root_dir` pattern defined in each server's configuration.  For [pyright](CONFIG.md#pyright), this is any directory containing ".git", "setup.py",  "setup.cfg", "pyproject.toml", or "requirements.txt").
 
-```lua
-local lspconfig = require'lspconfig'
-lspconfig.gopls.setup{
-  root_dir = lspconfig.util.root_pattern('.git');
-}
-```
+Language servers require each project to have a `root` in order to provide completion and search across symbols that may not be defined in your current file, and to avoid having to index your entire filesystem on each startup.
 
-The [documentation](CONFIG.md) for each config lists default values and
-additional optional properties. For a more complicated example overriding 
-the `name`, `log_level`, `message_level`, and `settings` of texlab:
-
-```lua
-local lspconfig = require'lspconfig'
-lspconfig.texlab.setup{
-  name = 'texlab_fancy';
-  settings = {
-    latex = {
-      build = {
-        onSave = true;
-      }
-    }
-  }
-}
-```
+## Enabling additional language servers
 
-### Example: custom config
-
-To configure a custom/private server, just 
-
-1. load the lspconfig module: `local lspconfig = require('lspconfig')`,
-2. Define the config: `lspconfig.configs.foo_lsp = { … }`
-3. Call `setup()`: `lspconfig.foo_lsp.setup{}`
-
-```lua
-local lspconfig = require'lspconfig'
-local configs = require'lspconfig/configs'
--- Check if it's already defined for when reloading this file.
-if not lspconfig.foo_lsp then
-  configs.foo_lsp = {
-    default_config = {
-      cmd = {'/home/ashkan/works/3rd/lua-language-server/run.sh'};
-      filetypes = {'lua'};
-      root_dir = function(fname)
-        return lspconfig.util.find_git_ancestor(fname) or vim.loop.os_homedir()
-      end;
-      settings = {};
-    };
-  }
-end
-lspconfig.foo_lsp.setup{}
-```
+Enabling most language servers is as easy as:
```

Comment:
I would say:

Enabling most language servers is as easy as installing that server globally, and configuring its setup:

Reply:
Thanks, I'll make that edit.

Reply:
Done